src/System.Data.Common/src/System/Data/Common/DbDataReaderExtensions.netcoreapp.cs

```diff
@@ -0,0 +1,75 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Data.Common
+{
+    public static partial class DbDataReaderExtensions
+    {
+        public static bool GetBoolean(this DbDataReader reader, string name) => reader.GetBoolean(reader.GetOrdinal(name));
```

Comment:
@bartonjs, @terrajobst, what's our current stance on validating the `this` argument in extension methods with an explicit check-for-null-and-throw-ArgumentNullException vs just letting it null ref?

Reply:
FDG doesn't have an opinion.  (Having an opinion is now on my list)

* Enumerable seems to be pretty consistent with ArgumentNullException
* MemoryExtensions seems to have either ArgumentNullException or "make null work".

Ergo I'm inferring our position is something like

* DO NOT allow the `this` parameter being `null` produce a `NullReferenceException`.
  * Extension methods are really static methods, and we generally don't like NullReferenceException from dereferencing parameters.
* AVOID defining extension methods that accept a `null` value for a `this` parameter.
  * It surprises people when typed `null` can be a valid "instance invocation".  But sometimes it makes sense.

Reply:
@bartonjs, shall I add the `null` check for these methods? If yes, is it by using `NotNullAttribute` or `reader is null` check?

Reply:
I've not seen the NotNull attribute used anywhere i'd say go with the is null check.

Reply:
Agreed.

Reply:
done